dips/dip-158.md

```diff
@@ -0,0 +1,795 @@
+|action|PaymentActionObject|Y|Information regarding the type of payment and payment amount|
+|reference_id|str|Y|Unique reference ID of this payment. This value should be globaly unique|
+|description|str|N|Description of the payment. To be displayed to the user|
+|
+
+```
+"payment_info": 
+{
+  "receiver": {
+    "address": "dm1pllhdmn9m42vcsamx24zrxgs3qqqpzg3ng32kvacapx5fl",         
+    "merchant_data": { 
+      "name": "Acme",
+      "legal_name": "Acme Technologies Inc.",
+      "address": {
+        "city": "San Francisco",
+        "country": "US",
+        "line1": "1260 Market Street",
+        "line2": "Suite 450",
+        "postal_code": "94103",
+        "state": "CA"
+      }        
+    }        
+  },
+  "action": {
+    "amount": "100000000",
+    "currency": "XUS",
+    "action": "charge",
+    "timestamp": "72322"
+  },
+  "reference_id": "4185027f-0574-6f55-2668-3a38fdb5de98",
+  "description": "payment request description"     
+}
+```
+
+## MerchantDataObject (new)
+The details of the merchant.
+|Field|Type|Required?|Description|
+|-|-|-|-|
+|name|str|Y|Merchant's display name. Should be recognizable by the payer|
+|legal_name|str|Y|The legal entity name|
+|address|AddressObject|Y|Merchant's physical address information|
+|
+```
+"merchant_data": 
+{
+  "name": "Acme",
+  "legal_name": "Acme Technologies Inc.",
+  "address": {
+    AddressObject()
+  }        
+}
+``` 
+
+## PayerDataObject (new)
+The details of the payer. payer_data is the minimal set of payer information required by the merchant to perform risk checks and apply AML policies.  
+|Field|Type|Required?|Description|
+|-|-|-|-|
+|given_name|str|Y|Legal given name of the payer|
+|surname|str|Y|Legal surname of the payer|
+|address|AddressObject|Y|Merchant's physical address information|
+|national_id|NationalIdObject|N|National ID information for payer. e.g. SSN|
+|
+```
+"payer_data": 
+{
+  "name": "Alice",
+  "surname": "Cooper",
+  "address": {
+    AddressObject()
+  },
+  "national_id": {
+    NationalIdObject()
+  }        
+}
+``` 
+---
+## Sender / Receiver Interface
+---
+
+This section describes a set of functionalities / activities that should be made available by a Sender and Receiver VASPs in order to support P2M flows based on this DIP. Each functionality can be conceptually compared to an API endpoint. Similar to API documentation, it includes the expected input and output for each functionality.
+
+| Functionality |Direction| Description | Command Type | Request Data | Response Data |
+|-|-|-|-|-|-|
+|GET_INFO|Sender > Receiver|By providing the object reference id the wallet can retrieve the object information from the receiving VASP. In most cases, this would be the first command sent by the wallet. The wallet is not expected to provide any payer data at this stage|InfoCommand|Reference ID|Payment Details ; Merchant Data|
```

Comment:
Instead of defining things twice (Functionality / Command Type) can we have 1 column?

Reply:
I was afraid you would say that :)) 
I actually considered removing it before creating the pull request.
The reason I kept it, is that it creates some separation between the functionality itself and the structure that currently supports it (which is the command). This might be useful when a command type is reused for different functionality. 
Having said that, this might be me being idealistic of me :) and if you feel strongly about it, I will remove the functionality column.